charts/spire/charts/spire-server/templates/tests/test-connection.yaml

```
@@ -17,4 +17,12 @@ spec:
       args: ['-zvw3', '{{ include "spire-server.fullname" . }}', '{{ .Values.service.port }}']
       securityContext:
         {{- toYaml .Values.securityContext | nindent 8 }}
+    {{- if eq (.Values.federation.enabled | toString) "true" }}
+    - name: wget-federation-bundle-endpoint
+      image: busybox
+      command: ['wget']
+      args: ['--no-check-certificate', '-O', '/dev/null', 'https://{{ include "spire-server.fullname" . }}.{{ .Release.Namespace }}.svc.cluster.local:{{ .Values.federation.bundleEndpoint.port }}']
```

A recent merge on upstream added the ability to override namespaces, below suggestion reuses that logic via this Macro to ensure this also works when the namespace is overridden.

```suggestion
      args: ['--no-check-certificate', '-O', '/dev/null', 'https://{{ include "spire-server.fullname" . }}.{{ include "spire-server.namespace" . }}.svc.cluster.local:{{ .Values.federation.bundleEndpoint.port }}']
```

---

This is a good suggestion.